src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -0,0 +1,852 @@
+        rowCells.push(this.expansionButton(rowId));
+        rowAccordionContent.push(this.rowAccordionContent(bodyRow.nestedContent));
+      } else {
+        rowCells.push(
+          <div
+            class={`${DATA_TABLE_CLASSES.CELL} ${DATA_TABLE_CLASSES.EXPANDABLE}`}
+            role="cell"
+          />
+        );
+      }
+    }
+
+    if (this.config.selectable) {
+      rowCells.push(this.selectRowCheckbox(false, rowId));
+    }
+
+    let cellIndex = 0;
+
+    bodyRow.data.forEach((rowCell: any) => {
+      const columnSettings = this.data.head[
+          Object.keys(this.data.head)[cellIndex]
+        ],
+        alignment = this.cellAlignment(
+          rowCell.align ? rowCell.align : columnSettings.align || null
+        ),
+        cellLabel = rowCell.label || columnSettings.label,
+        cellWidth =
+          this.config.columnSizes && this._currentScreenBreakpoint
+            ? this.config.columnSizes[this._currentScreenBreakpoint][
+                cellIndex
+              ]
+            : null;
+      let cellData;
+
+      if (!!rowCell.template && !!this.$scopedSlots[rowCell.template]) {
+        if (typeof rowCell === 'object' && rowCell.payload) {
+          cellData = this.$scopedSlots[rowCell.template]!(rowCell.payload);
+        }
+      } else if (typeof rowCell === 'object' && !!rowCell.value) {
+        cellData = (
+          <Tooltip message={rowCell} class="-w--100">
+            <div class={UTILITY_CLASSES.TYPOGRAPHY.TEXT_TRUNCATE}>
+              {rowCell.value}
+            </div>
+          </Tooltip>
+        );
+      } else if (
+        typeof rowCell === 'string' ||
+        typeof rowCell === 'number') {
+        cellData = (
+          <Tooltip message={rowCell} class="-w--100">
+            <div class={UTILITY_CLASSES.TYPOGRAPHY.TEXT_TRUNCATE}>
+              {rowCell}
+            </div>
+          </Tooltip>
+        );
+      } else {
+        cellData = null;
+      }
+
+      rowCells.push(
+        <div
+          aria-label={cellLabel}
+          class={`
+            ${DATA_TABLE_CLASSES.CELL}
+            ${alignment}
+            ${cellWidth && cellWidth > 0 ? `-flex-basis--${cellWidth}` : ''}`}
+          data-label={cellLabel}
+          role="cell"
+          style={`
+            ${cellWidth === 0 ? 'display: none' : ''}
+            ${columnSettings.allowOverflow ? 'overflow: visible;' : ''}
+            `}
+        >
+          {cellData}
+        </div>
+      );
+
+      cellIndex++;
+    });
+
+    row.push(
+      <div
+        id={rowId}
+        class={`
+        ${rowClass}
+        ${striped ? DATA_TABLE_CLASSES.STRIPED : ''}
+        ${this.selectedRows.includes(rowId) ? ACTIVE_CLASS : ''}
+        ${this.accordionsShown.includes(rowId) ? EXPANDED_CLASS : ''}
+        `}
+        role="row"
+      >
+        {rowCells}
+      </div>
+    );
+
+    if (bodyRow.nestedContent) {
+      row.push(
+        <transition name="slide-fade">
+          <div
```

Comment:
Does the table listens to row clicks and emits the event with the selected row data?.
or the component consumer can configure it?

Reply:
Yes, row selection emits `chiSelectedRowsChange` event with the corresponding data of all the selected rows.